test/Align/test_align.py

```diff
@@ -364,25 +364,54 @@ def test_hydrogen_mass_repartitioning():
     # Create the merged molecule.
     merged = BSS.Align.merge(m0, m1, mapping, allow_ring_breaking=True)
 
+    # Create a dummy element.
+    dummy = Element("Xx")
+
+    # Get the elements in either end state.
+    elements0 = merged._sire_object.property("element0").toVector()
+    elements1 = merged._sire_object.property("element1").toVector()
+
     # Work out the initial mass of the system.
     initial_mass0 = 0
-    for mass in merged._sire_object.property("mass0").toVector():
-        initial_mass0 += mass.value()
+    for idx, mass in enumerate(merged._sire_object.property("mass0").toVector()):
+        if elements0[idx] != dummy:
+            initial_mass0 += mass.value()
     initial_mass1 = 0
-    for mass in merged._sire_object.property("mass1").toVector():
-        initial_mass1 += mass.value()
+    for idx, mass in enumerate(merged._sire_object.property("mass1").toVector()):
+        if elements1[idx] != dummy:
+            initial_mass1 += mass.value()
 
     # Repartition the hydrogen mass.
     merged.repartitionHydrogenMass()
 
+    # Lists to store the mass of dummy atoms in the two end states.
+    dummy_masses0 = []
```

Comment:
Similarly, `sire` might have a function that will give you the dummy and non-dummy lists as a one-liner?

Reply:
Yes, you can now do things like `mol["not element Xx"].property("mass")`, but this doesn't seem to work for version of Sire used for this PR. Also, extracting elements for a non-standard property name (as in this case) doesn't seem to work at present.